dev-python/protobuf-python/protobuf-python-3.20.1-r1.ebuild

```diff
@@ -4,58 +4,61 @@
 EAPI=8
 
 DISTUTILS_USE_PEP517=setuptools
-PYTHON_COMPAT=( python3_{8..10} )
+PYTHON_COMPAT=( python3_{8..11} )
 
 inherit distutils-r1
 
+PARENT_PN="${PN/-python/}"
+PARENT_PV="${PV}"
+PARENT_P="${PARENT_PN}-${PARENT_PV}"
+
 if [[ "${PV}" == "9999" ]]; then
 	inherit git-r3
 
-	EGIT_REPO_URI="https://github.com/protocolbuffers/protobuf"
+	EGIT_REPO_URI="https://github.com/protocolbuffers/${PARENT_PN}"
 	EGIT_SUBMODULES=()
+	EGIT_CHECKOUT_DIR="${WORKDIR}/${PARENT_P}"
+else
+	SRC_URI="
+		https://github.com/protocolbuffers/${PARENT_PN}/archive/v${PARENT_PV}.tar.gz
+			-> ${PARENT_P}.tar.gz
+	"
+	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~x64-macos"
 fi
 
 DESCRIPTION="Google's Protocol Buffers - Python bindings"
 HOMEPAGE="
 	https://developers.google.com/protocol-buffers/
-	https://github.com/protocolbuffers/protobuf/
+	https://github.com/protocolbuffers/protobuf
 	https://pypi.org/project/protobuf/
 "
-if [[ "${PV}" != "9999" ]]; then
-	SRC_URI="
-		https://github.com/protocolbuffers/protobuf/archive/v${PV}.tar.gz
-			-> protobuf-${PV}.tar.gz
-	"
-fi
-S="${WORKDIR}/protobuf-${PV}/python"
 
 LICENSE="BSD"
-SLOT="0/30"
-KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~x64-macos"
+SLOT="0/31"
+
+S="${WORKDIR}/${PARENT_P}/python"
 
 BDEPEND="
 	${PYTHON_DEPS}
-	~dev-libs/protobuf-${PV}
+	dev-libs/protobuf:${SLOT}
 	dev-python/six[${PYTHON_USEDEP}]
 "
 DEPEND="
 	${PYTHON_DEPS}
-	~dev-libs/protobuf-${PV}
+	dev-libs/protobuf:${SLOT}
 "
 RDEPEND="
 	${BDEPEND}
-	!dev-python/namespace-google
 "
 
-if [[ "${PV}" == "9999" ]]; then
-	EGIT_CHECKOUT_DIR="${WORKDIR}/protobuf-${PV}"
-fi
-
 distutils_enable_tests setup.py
 
+PATCHES=(
```

Comment:
If array is empty, just remove?

Reply:
If we compare with the 3.19.3 ebuild, we notice the former was not using the default mechanism. With this explicit array, we know where we need to put patches if required. Like a default template.

Okay it’s empty, but for now on.

Unless this has an impact on a behavior I am not aware, I think it’s worth keeping it in plain sight.